examples.md

```diff
@@ -108,3 +109,12 @@ uses: actions/cache@preview
     restore-keys: |
       ${{ runner.os }}-go-
 ```
+
+## Haskell - Cabal
+
+```yaml
+- uses: actions/cache@preview
+  with:
+    path: ~/.cabal/store
+    key: ${{ runner.os }}-haskell-${{ matrix.ghc }}
```

Comment:
What does `matrix.ghc` represent?

Typically we want the key to contain some reference to the dependencies so that we get a new key and cache when dependencies change (see the other examples with `hashFiles`).

Reply:
> What does matrix.ghc represent?

the version of the GHC compiler used; with Haskell it's customary to test various versions of the compiler, whose cached artifacts are incompatible to each other, hence using the GHC version as cache-index/key

Fwiw, the concept of "lockfiles" is a different paradigm from the Nix-style store (which is really a hash-indexed cache already) used by cabal.

Reply:
Yep, you have two main options.

If the path is constant under the home directory, you can use `~` which the action will resolve to `$HOME` on macOS/Linux and `%USERPROFILE%` on Windows.

If it's more complicated, you can use an extra step beforehand to determine the cache location and output that. See https://github.com/actions/cache/blob/master/examples.md#php---composer for an example of that.

Reply:
Yeah, using %USERPROFILE% on windows is not a good idea, because it's prone to hitting the anachronistic PATH_MAX limits much more likely; so we we typically set it to something like `C:\SR` to workaround the limit until all tooling is able to safely use more recent Windows' long paths APIs.

The composer approach sounds great; maybe long-term the `setup-haskell` action could provide such an output variable (unfortunately I lack the JavaScript skills to teach `setup-haskell` various things I've already inlined into a workflow file...)

Thanks for the pointer!

Reply:
I believe the hosted windows runners should have long path support enabled when initialized via:
```powershell
# See https://docs.microsoft.com/en-us/windows/desktop/fileio/naming-a-file#maximum-path-length-limitation
Set-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem' -Name 'LongPathsEnabled' -Value 1
```

>maybe long-term the setup-haskell action could provide such an output variable
That's a great idea!

>unfortunately I lack the JavaScript skills to teach setup-haskell various things I've already inlined into a workflow file.

Feel free to file an issue https://github.com/actions/setup-haskell, I'm sure the feedback would be welcomed!